### PR TITLE
Resource manifest source generator

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

File: src/OrchardCore/OrchardCore.SourceGenerator.Resources/ResourceManfiestGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace OrchardCore.SourceGenerator.Resources
+{
+    [Generator]
+    internal class ResourceManfiestGenerator : ISourceGenerator
```
Comment (Skrypt) [inline src/OrchardCore/OrchardCore.SourceGenerator.Resources/ResourceManfiestGenerator.cs:12]: ResourceManfiestGenerator => ResourceManifestGenerator

File: src/OrchardCore/OrchardCore.SourceGenerator.Resources/ResourceManfiestGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    {
+        private static readonly string _resourcesFile = "resources.json";
+
+        public void Execute(GeneratorExecutionContext context)
+        {
+            var sourceBuilder = new StringBuilder(@"
+using System;
```
Comment (Skrypt) [inline src/OrchardCore/OrchardCore.SourceGenerator.Resources/ResourceManfiestGenerator.cs:19]: This looks like a bad copy/paste code block

File: src/OrchardCore/OrchardCore.SourceGenerator.Resources/ResourceManfiestGenerator.cs (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore/OrchardCore.SourceGenerator.Resources/ResourceManfiestGenerator.cs:12]: Nice catch

**Comment by sebastienros** [comment]
How is that better?

**Comment by hishamco** [comment]
> How is that better?

This is still WIP, the idea is that we don't need to update the code for the resource manifest everytime a resource library has been released or upgraded. Moreover we could automatically generate SRI instead of doing that ourselves which is leads to issues sometime, which I already push 2 PRs related

This way nothing we need except add an update in a `resources.json` then the magic will happen using source generators ;)

**Comment by Piedone** [comment]
Is this something you'd like to revisit any time soon @hishamco or should we close?

FYI we recently did something similar (WIP but mostly done): https://github.com/Lombiq/Helpful-Libraries/issues/238

**Comment by hishamco** [comment]
> FYI we recently did something similar (WIP but mostly done): https://github.com/Lombiq/Helpful-Libraries/issues/238

Oh really I will have a look ..

**Comment by hishamco** [comment]
This also could be done using MS Build task #12364

**Comment by hishamco** [comment]
> FYI we recently did something similar (WIP but mostly done): https://github.com/Lombiq/Helpful-Libraries/issues/238

In my case, there is no need for `ConstantFromJson` because all the styles & scripts are in one JSON file, in my case `resources.json`. I will update this ASAP, also it's good to demo it along side with msbuild tasks

**Comment by hishamco** [comment]
I'm not sure why `SubResourceIntegrityTests` complains, we might need to revise the unit test

**Comment by Piedone** [comment]
FYI here's a Source Generator implementation that works all the time in Visual Studio too: https://github.com/Lombiq/Helpful-Libraries/pull/251

**Comment by hishamco** [comment]
While the PR is not interested by Seb :) can I ask why we need to keep track of all the versions for each style or script that we used from the beginning?

**Comment by sebastienros** [comment]
Sorry @hishamco closing because it would be more churn to fix than it's adding value. Thanks for the effort.

**Pull request closed without merging**